### ISSUE TITLE
Epub.js without pageMap and CFİ

### ISSUE THREAD

**Issue by ozibrahim** [futurepress/epub.js#259]
Hi, i am trying to develop cross platform epub reader. Our epub's some chapters includes more than 1.000 p nodes in body. So renderer.js mapPage function takes nearly 3 seconds. (Samsung T530 Android Tablet / Chrome)
Can i decrease this time, then how?

On the other hand, i think page determination logic should based on page's ratio in chapter and page's chapter. And, i guess we don't need epubcfi's actually, so can i remove mapPage function and edit pageMap related codes?

To sum up, how can we increase performance?

Thanks,

**Comment by fchasen** [comment]
I think making pageMap optional makes sense, it really only helps to do it before rendering if there are going to changes to the content (such as adding highlights).

**Comment by fchasen** [comment]
Fixed in v0.3, mapping happens for only visible content when location changes now.
